### PR TITLE
Feature/nex 769/oidc endpoints

### PR DESCRIPTION
**Pull request by ekkinox**
see https://oat-sa.atlassian.net/browse/NEX-769

- added OIDC init endpoint (for tools)
- added related tests

### CODE REVIEW

File: Tests/Functional/Action/Tool/OidcLoginInitiationActionTest.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+{
+    /** @var KernelBrowser */
+    private $client;
+
+    /** @var RegistrationInterface */
+    private $registration;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
```
Comment (emgolubev) [inline Tests/Functional/Action/Tool/OidcLoginInitiationActionTest.php]: can be removed

File: Tests/Functional/Action/Tool/OidcLoginInitiationActionTest.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        $this->client = static::createClient();
+
+        $this->registration = static::$container
+            ->get(RegistrationRepositoryInterface::class)
+            ->find('testRegistration');
+    }
+
+    public function testValidLoginInitiationWithPostMethod(): void
+    {
+        $this->client->request(Request::METHOD_POST, '/oidc/login-initiation', [
```
Comment (emgolubev) [inline Tests/Functional/Action/Tool/OidcLoginInitiationActionTest.php]: Optional. I would like to do the same code formatting as you did for the `GET` method test.

File: Tests/Functional/Action/Tool/OidcLoginInitiationActionTest.php (hunk 1 above)
Comment (ekkinox) [reply Tests/Functional/Action/Tool/OidcLoginInitiationActionTest.php]: done

File: Tests/Functional/Action/Tool/OidcLoginInitiationActionTest.php (hunk 2 above)
Comment (ekkinox) [reply Tests/Functional/Action/Tool/OidcLoginInitiationActionTest.php]: done

**Review by emgolubev — APPROVED** [review]

**Review by m-lagarde — APPROVED** [review]

**Review by peetya — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Code is self documented

**Pull request merged**